src/basic/pidref.c

```diff
@@ -260,6 +260,12 @@ int pidref_verify(PidRef *pidref) {
         return 1; /* We have a pidfd and it still points to the PID we have, hence all is *really* OK → return 1 */
 }
 
+int pidref_is_alive(PidRef *pidref) {
+        assert(pidref);
+
+        return pidref->fd >= 0 ? pidfd_is_alive(pidref->fd) : pid_is_alive(pidref->pid);
```

Comment:
Here, if `pidfd_is_alive` is called, return value would be `-ESRCH` if process is already dead, but for `pid_is_alive` `false` is returned.